src/main/java/org/stellar/sdk/Server.java

```diff
@@ -28,28 +29,36 @@
      */
     private static final int HORIZON_SUBMIT_TIMEOUT = 60;
 
-    public Server(String uri) {
-        this(uri,
-                new OkHttpClient.Builder()
-                    .addInterceptor(new ClientIdentificationInterceptor())
-                    .connectTimeout(10, TimeUnit.SECONDS)
-                    .readTimeout(30, TimeUnit.SECONDS)
-                    .retryOnConnectionFailure(true)
-                    .build(),
-                new OkHttpClient.Builder()
-                    .addInterceptor(new ClientIdentificationInterceptor())
-                    .connectTimeout(10, TimeUnit.SECONDS)
-                    .readTimeout(HORIZON_SUBMIT_TIMEOUT + 5, TimeUnit.SECONDS)
-                    .retryOnConnectionFailure(true)
-                    .build()
+    public Server(String uri, Network network) {
+        this(
+            uri,
+            new OkHttpClient.Builder()
+                .addInterceptor(new ClientIdentificationInterceptor())
+                .connectTimeout(10, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .retryOnConnectionFailure(true)
+                .build(),
+            new OkHttpClient.Builder()
+                .addInterceptor(new ClientIdentificationInterceptor())
+                .connectTimeout(10, TimeUnit.SECONDS)
+                .readTimeout(HORIZON_SUBMIT_TIMEOUT + 5, TimeUnit.SECONDS)
+                .retryOnConnectionFailure(true)
+                .build(),
+            network
         );
 
     }
 
-    public Server(String serverURI, OkHttpClient httpClient, OkHttpClient submitHttpClient) {
+    public Server(
+            String serverURI,
+            OkHttpClient httpClient,
+            OkHttpClient submitHttpClient,
+            Network network
```

Comment:
I'm wondering if, instead, we should request `/` (`Server.root()`) that exposes `network_passphrase` value. Not in a constructor but, ex. once in `submitTransaction`. It's simpler and prevents situations where users mistakenly set a network passphrase to a wrong one and receive bad signature errors for their transactions.

Reply:
@bartekn that could potentially be interesting to make the server instance self configuring where it initializes by doing a request to Server.root(). That could make setting up a standalone network simpler.